### DISCUSSION TITLE
srsRAN Project with srsUE and ZMQ ping does not go through

### DISCUSSION THREAD

**Discussion by oguztudo** [srsran/srsRAN_Project#81] [General Help] [answered]
Hello everybody,

I am currently trying to make the gNB work with ZMQ. So far i downloaded the configuration files that were given in 
https://docs.srsran.com/projects/project/en/latest/tutorials/source/srsUE/source/index.html#zeromq-based-setup

Since i could not utilize the 10.x.x.x IP addresses in the config files I edited the addresses of the amf so the gNB could connect to the sctp socket (everything else is the same):
- amf:
-   addr: 127.0.0.5
-   bind_addr: 127.0.0.1

The connection can be established as can be seen in Screenshot one. When I ping the assigned IP 10.45.0.10 the terminal output of the gNB shows traffic on the DL side, however the ping ends in a 100% packet loss (Screenshot 2).

On the UL side i tried various possible IP-addresses such as the IP of the UE, the bind_addr, localhost, just antything that could be a possibility since nothing seems to have worked for me. But theres simply no possibility to ping with "sudo ip netns ue1 exec x.x.x.x".

All files that have been used have beend added as attachments. The ping process can be seen in the log quite in the beginning and at around 15:49:40.
[gnb_zmq.txt](https://github.com/srsran/srsRAN_Project/files/11518859/gnb_zmq.txt)
[ue_zmq.txt](https://github.com/srsran/srsRAN_Project/files/11518860/ue_zmq.txt)
[ue.log](https://github.com/srsran/srsRAN_Project/files/11518865/ue.log)
[gnb.log](https://github.com/srsran/srsRAN_Project/files/11518893/gnb.log)

Did someone encounter the same issue and can provide any help? 

<img width="861" alt="UE_connected" src="https://github.com/srsran/srsRAN_Project/assets/129198919/c64a0b30-6f7a-465d-923f-679b31216b73">
<img width="990" alt="Ping" src="https://github.com/srsran/srsRAN_Project/assets/129198919/182deefa-f502-4834-a0f9-92e5383a3f6b">

**Comment by ismagom** [comment, marked as answer]
Hi, thanks for the feedback. 

Are you using the latest commit? I see something weird in the gnb.log: 

```
2023-05-19T15:46:47.515345 [PDCP    ] [I] ue=0 DRB1 DL: PDCP configured. rb_type=DRB rlc_mode=AM sn_size=18 discard_timer=-1 count_notify=3221225472 count_max=4294967294
2023-05-19T15:46:47.515347 [PDCP    ] [I] ue=0 DRB1 UL: PDCP configured. rb_type=DRB rlc_mode=AM sn_size=18 t_reordering=0 count_notify=3221225472 count_max=3489660928
2023-05-19T15:46:47.515574 [RLC     ] [I] ue=0 DRB1 DL: RLC UM configured. tx_sn_size=12
2023-05-19T15:46:47.515577 [RLC     ] [I] ue=0 DRB1 UL: RLC UM configured. rx_sn_size=12 t_reassembly=50
```
It seems it's activating RLC UM mode, when it should use AM by default. Have you changed the 5QI settings in the gnb or open5gs? If you haven't, try pulling the latest code and recompile. 

Thanks

**Reply by oguztudo** [reply]
Hello ismagom and thanks for the help.
That actually did the trick for me. My build was actually from when the ZMQ update appeared but apparently something went wrong there.
Now the ping packets are being sent and received perfectly fine and i get changes in DL and UL as well :)
